Push down subquery before using COUNT/COUNT_BIG if the query has an offset (fixes #7523)

tuespetre opened the pull request:
🌞

dnfclas commented:
Hi __@tuespetre__, I'm your friendly neighborhood .NET Foundation Pull Request Bot (You can call me DNFBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by .NET Foundation and real humans are currently evaluating your PR.</p>

TTYL, DNFBOT;

tuespetre commented:
:up::date:

I copied the two tests and added the ordering to the copies so both cases will be covered 'on the record'. I hope that's alright. I also added the overrides for all four tests in `RowNumberPagingTest`.

smitpatel commented:
Merged via eddbfa2a377a72f2330266a9f0881ac172736a19
